standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2109,7 +2109,8 @@ struct RenamePartitionRequest {
   3: required string tableName,
   4: required list<string> partVals,
   5: required Partition newPart,
-  6: optional string validWriteIdList
+  6: optional string validWriteIdList,
+  7: optional EnvironmentContext environmentContext
```

Comment:
Would it be better to add the flag as a parameter instead hiding in the environmentContext?

Reply:
Maybe also add the `txnId` too?

Reply:
sure

Reply:
By my understanding - EnvironmentContext was the first solution for extending the interface with new fields without adding new methods every time. This solution evolved to the `...Request` objects later.

Since we have a `RenamePartitionRequest` already, it might be better to use that.

Reply:
RenamePartitionRequest is only used by thrift to pass the data, however, it's not used by the server-side. All its internals are extracted into individual params including `EnvironmentContext`.
HMSHandler.alterPartitionForTruncate, HiveAlterHandler.alterPartition do not operate on request object, but individual params. 
If we want to use RenamePartitionRequest that would require a major refactor. Currently, if I would simply extend RenamePartitionRequest with `txnid` and`RENAME_MAKE_DATA_COPY` I would still need to pass them through the `EnvironmentContext` on a server-side.